In [1]:
from scipy import stats
import numpy as np

__Задача 1__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

__Решение__

(Изначально предположим что здесь нормальное распределение)

Используем следующую статистику:

$$T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}$$

Подставляем значения:

$$T(X) = \dfrac{\overline{X} - 17}{2 / 10}$$

Пока не используем данные из выборки кроме ее кол-ва

Критическая область: $\Omega_\alpha = (t_{1 - \alpha}, \infty)$


In [11]:
t = stats.norm.ppf(1-0.025)
t

1.959963984540054

Получаем: $\Omega_\alpha = ({1.959}, \infty)$

Теперь остаётся лишь взять среднее по выборке из усовия задачи, посчитать для неё значение функции $T$ и проверить, содержится ли оно в критической области.

In [13]:
mean = 17.5

T = (mean - 17)/(2/10)
T

2.5

Итак, значение __НЕ__ попало в критическую область, соответственно мы принимаем нулевую гепотезу о том, что средний диаметр равняется 17мм


__Задача 2__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания __a__ с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

__Решение__

Воспользуемся выше приведенной статистикой:

$$T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}$$

Надежность:
$$p = 0.95$$

Соответсвенно имеем:

$$
P \left(
    \overline{X} - t_{1 - 0.95/2} \cdot \dfrac{\sigma}{\sqrt{n}}
    \leq a \leq 
    \overline{X} + t_{1 - 0.95/2} \cdot \dfrac{\sigma}{\sqrt{n}}
\right) = p.
$$

$$
P \left(
    \overline{X} - t_{1 - 0.95/2} \cdot \dfrac{16}{\sqrt{256}}
    \leq a \leq 
    \overline{X} + t_{1 - 0.95/2} \cdot \dfrac{16}{\sqrt{256}}
\right) = p.
$$

In [3]:
alpha = 0.05
t = stats.t.ppf(1 - alpha / 2, df=256)
t

1.9692738896550188

In [4]:
mean = 80
std = 16
n = 256

Получаем доверительный интервал:

In [5]:
(mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n))

(78.03072611034499, 81.96927388965501)


__Задача 3__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

__Решение__

In [6]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [7]:
mean = samples.mean()
mean

198.5

Имеем:

уровень значимости __a__ = 0.1

$\overline{X}$ = 198.5

и __n__ = 10

__неизвестна лишь сигма!__

Так как мы не знаем дисперсию и распределение нормальное, воспользуемся __статистикой Стьюдента__:

$$T(X) = \dfrac{\overline{X} - a}{\sigma_X / \sqrt{n}}$$

Получаем критическую область:

$$\Omega_\alpha = \left( -\infty, - t_{1 - \alpha/2, \: n - 1} \right) \cup \left( t_{1 - \alpha/2, \: n - 1}, \infty \right).$$

In [8]:
std = samples.std(ddof=1)
std

4.453463071962462

In [9]:
t = stats.t.ppf(1 - 0.1 / 2, df=10-1)
t

1.8331129326536335

$$\Omega_\alpha = \left( -\infty, - 1.833 \right) \cup \left( 1.833, \infty \right)$$

In [10]:
T = (mean - 200) / (std / np.sqrt(10))
T

-1.0651074037450896

Итак, значение __НЕ__ попало в критическую область, соответственно мы принимаем нулевую гепотезу о том, что средний вес пачки печенья 200 г.